src/java.desktop/share/classes/com/sun/media/sound/JavaSoundAudioClip.java

```diff
@@ -102,14 +103,28 @@ public static JavaSoundAudioClip create(final File file) throws IOException {
         return clip;
     }
 
-    public static JavaSoundAudioClip create(final URLConnection uc) {
-        JavaSoundAudioClip clip = new JavaSoundAudioClip();
+    /* Used [only] by sun.awt.www.content.MultiMediaContentHandlers */
+    public static SoundClip create(final URLConnection uc) {
+        File tmpFile = null;
         try {
-            clip.init(uc.getInputStream());
-        } catch (final Exception ignored) {
-            // Playing the clip will be a no-op if an exception occured in inititialization.
+            tmpFile = File.createTempFile("javaurl", ".aud");
+        } catch (IOException e) {
+            return null;
         }
-        return clip;
+
+        try (InputStream in = uc.getInputStream();
+             FileOutputStream out = new FileOutputStream(tmpFile)) {
+             in.transferTo(out);
+        } catch (IOException e) {
+        }
+
+        try {
+             return SoundClip.createSoundClip(tmpFile);
+        } catch (IOException e) {
```

Comment:
```suggestion
        } catch (IOException ignored) {
```
To explicitly document that we ignore the exception, and to avoid warnings in IDE.